android-o/AutofillFramework/AutofillFramework/app/CreditCardActivity.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Linq;
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Support.V7.App;
+using Android.Widget;
+using Java.Lang;
+using Java.Util;
+
+namespace AutofillFramework.app
+{
+	[Activity(Label = "CreditCardActivity")]
+	public class CreditCardActivity : AppCompatActivity
+	{
+		static int CC_EXP_YEARS_COUNT = 5;
+
+		string[] Years = new string[CC_EXP_YEARS_COUNT];
+
+    	Spinner mCcExpirationDaySpinner;
+		Spinner mCcExpirationMonthSpinner;
+		Spinner mCcExpirationYearSpinner;
+		EditText mCcCardNumber;
+
+		public static Intent GetStartActivityIntent(Context context)
+		{
+			var intent = new Intent(context, typeof(CreditCardActivity));
+        	return intent;
+    	}
+
+		class YearAdapter : ArrayAdapter
+		{
+			string[] Years { get; set; }
+			public YearAdapter(Context context, string[] years) : base(context, Android.Resource.Layout.SimpleSpinnerItem, years)
+			{
+				Years = years;
+			}
+
+			public override ICharSequence[] GetAutofillOptionsFormatted()
+			{
+				// convert C# string into Java String to return a CharSequence array
+				var javaStringList = new Java.Lang.String[Years.Count()];
+				for (var pos = 0; pos < Years.Count(); pos++) {
+					javaStringList[pos] = new Java.Lang.String(Years[pos]);
+				}
+				return javaStringList;
+			}
+		}
+
+		protected override void OnCreate(Bundle savedInstanceState)
+		{
+			base.OnCreate(savedInstanceState);
+			SetContentView(Resource.Layout.credit_card_activity);
+			mCcExpirationDaySpinner = (Android.Widget.Spinner)FindViewById(Resource.Id.expirationDay);
+			mCcExpirationMonthSpinner = (Android.Widget.Spinner)FindViewById(Resource.Id.expirationMonth);
+			mCcExpirationYearSpinner = (Android.Widget.Spinner)FindViewById(Resource.Id.expirationYear);
+			mCcCardNumber = (Android.Widget.EditText)FindViewById(Resource.Id.creditCardNumberField);
```

Comment:
Android.Widget declarations above are unnecessary due to previous using declaration.

Reply:
Done

Reply:
Done